src/Common/tests/System/Net/Configuration.Certificates.cs

```diff
@@ -18,78 +16,56 @@ public static partial class Certificates
+                        byte[] clientCertificateBytes = File.ReadAllBytes(Path.Combine(TestDataFolder, "testclienteku.contoso.com.pfx"));
+                        byte[] noEKUCertificateBytes = File.ReadAllBytes(Path.Combine(TestDataFolder, "testnoeku.contoso.com.pfx"));
+                        byte[] selfSignedServerCertificateBytes = File.ReadAllBytes(Path.Combine(TestDataFolder, "testselfsignedservereku.contoso.com.pfx"));
+                        byte[] selfSignedClientCertificateBytes = File.ReadAllBytes(Path.Combine(TestDataFolder, "testselfsignedclienteku.contoso.com.pfx"));
 
-                try
-                {
-                    return new X509Certificate2(
-                        File.ReadAllBytes(Path.Combine(TestDataFolder, certificateFileName)),
-                        CertificatePassword,
-                        X509KeyStorageFlags.DefaultKeySet | X509KeyStorageFlags.Exportable);
-                }
-                catch (Exception ex)
-                {
-                    Debug.Fail(nameof(Configuration.Certificates.GetCertificate) + " threw " + ex.ToString());
-                    throw;
-                }
-                finally
-                {
-                    if (PlatformDetection.IsWindows)
+                        // On Windows, applications should not import PFX files in parallel to avoid a known system-level
+                        // race condition bug in native code which can cause crashes/corruption of the certificate state.
+                        Assert.True(mutex?.WaitOne(MutexTimeoutMs) ?? true, "Could not acquire the global certificate mutex.");
+                        try
+                        {
+                            s_serverCertificate = new X509Certificate2(serverCertificateBytes, CertificatePassword, X509KeyStorageFlags.Exportable);
+                            s_clientCertificate = new X509Certificate2(clientCertificateBytes, CertificatePassword, X509KeyStorageFlags.Exportable);
+                            s_noEKUCertificate = new X509Certificate2(noEKUCertificateBytes, CertificatePassword, X509KeyStorageFlags.Exportable);
+                            s_selfSignedServerCertificate = new X509Certificate2(selfSignedServerCertificateBytes, CertificatePassword, X509KeyStorageFlags.Exportable);
+                            s_selfSignedClientCertificate = new X509Certificate2(selfSignedClientCertificateBytes, CertificatePassword, X509KeyStorageFlags.Exportable);
+                        }
+                        finally { mutex?.ReleaseMutex(); }
+                    }
+                    catch (Exception ex)
                     {
-                        s_importPfxMutex.ReleaseMutex();
+                        Trace.Fail(nameof(Certificates) + " cctor threw " + ex.ToString());
+                        throw;
                     }
                 }
             }
+
+            public static X509Certificate2 GetServerCertificate() => new X509Certificate2(s_serverCertificate);
```

Comment:
These methods are making copies of the X509Certificate2 object because we assume that the caller of the Get*Certificate() methods ends up owning the certificate and probably does a 'Dispose' of it?  It might be nice to add a comment here to remind people of that.

Reply:
Yeah, all of the call sites that use these dispose them.  I can add a comment.